lib/bootloader_pvm.pm

```diff
@@ -179,7 +179,12 @@ sub boot_hmc_pvm {
     # don't wait for it, otherwise we miss the menu
     type_string "mkvterm -m $hmc_machine_name --id $lpar_id\n";
     # skip further preperations if system is already installed
-    return if get_var('BOOT_HDD_IMAGE');
+    # PowerVM, send "up" key to refresh the serial terminal if
+    # it is already entered into grub2 menu
+    if (get_var('BOOT_HDD_IMAGE')) {
+        send_key('up') if (!check_screen('grub2', 100));
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
At here, I only has one tag, if use assert_screen this tag cannot match then the case will fail.  If in 100s the grub page cannot show up. I will refresh the serial output by send key up.

Reply:
How about add a new needle here?

Reply:
No, It's not good. First we need build one new tag and needle. Second it doesn't have special needle at here. If don't need refresh the serial, it also has 'Open Complete' needle.

Reply:
We use check_screen('grub2', 100)), but we don't waste 100s at here. As grub page need wait about 100s to show. After this part, it will go to wait_boot -> wait_grub function, then it will wait grub again. As we have waited 100s at first, it will save time at wait_grub function. So I think check_screen('grub2', 100)) is good enough.

Reply:
As no matter we need refresh the serial or not, the needles all the same.  So we cannot use needle to check whether we need send up key or not. In case, we waste 100s to check grub page at here, We will send up key any way. It will not impact the other case. Please check the case  http://openqa.suse.de/t5802150 log, it also send up key even it doesn't need refresh serial output.